src/Stratis.Bitcoin.Features.Consensus.Tests/TestChainFactory.cs

```diff
@@ -128,8 +128,8 @@ public static async Task<TestChainContext> CreateAsync(Network network, string d
             testChainContext.PeerBanning = new PeerBanning(testChainContext.ConnectionManager, testChainContext.LoggerFactory, testChainContext.DateTimeProvider, testChainContext.PeerAddressManager);
             var deployments = new NodeDeployments(testChainContext.Network, testChainContext.Chain);
             testChainContext.ConsensusRules = new PowConsensusRuleEngine(testChainContext.Network, testChainContext.LoggerFactory, testChainContext.DateTimeProvider,
-                testChainContext.Chain, deployments, consensusSettings, testChainContext.Checkpoints, cachedCoinView, testChainContext.ChainState, new InvalidBlockHashStore(new DateTimeProvider()))
-                .Register();
+                testChainContext.Chain, deployments, consensusSettings, testChainContext.Checkpoints, cachedCoinView, testChainContext.ChainState,
+                    new InvalidBlockHashStore(new DateTimeProvider()), new NodeStats(new DateTimeProvider())).Register();
```

Comment:
`new DateTimeProvider()` is everywhere, could we perhaps create a common instance and use it

Reply:
It's ok. Why we need single instance in tests?

Reply:
Well, if it a unit test, we shouldn't be passing concrete implementations but rather mocks. Also easier to manage one instance in one place.

Reply:
Concrete implementations are better for the test's robustness. If it's easy to use concrete instance I'd always go with one.

Reply:
In general, you want to mock all dependencies so that your test focuses on testing the `unit` logic. It doesn't matter, don't want to start a whole debate here. If you want to keep it this way, fine.